script-injection.php

```diff
@@ -33,67 +33,125 @@ function inject_js() {
           echo do_category_js( $cookie_category ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
         } ?>
       }
 
-      airCookieSettings.onChange = function() {
+      function ccOnChange() {
         airCookierecordConsent();
-
         <?php foreach ( $cookie_categories as $cookie_category ) {
           echo do_category_js( $cookie_category ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
         } ?>
+        checkIframeConsent();
       }
-    <?php endif; ?>
 
-    <?php // Run the Cookie Consent at last. ?>
-    cc.run( airCookieSettings );
+      <?php // Function to detect and change regex cookies ?>
+      function checkRegexCookies(categories) {
+          for (let categoryName in categories) {
+              let category = categories[categoryName];
+              <?php // Check if the category has cookies specified ?>
+              if (category.autoClear) {
+                  category.autoClear.cookies.forEach(function(cookie) {
+                      <?php // Regex pattern for ^(*) ?>
+                      let cookie_regex_check = /\^\(.*\)/;
+                      if (cookie_regex_check.test(cookie.name) === true ) {
+                          <?php // Replace regular string with regex object (https://cookieconsent.orestbida.com/reference/configuration-reference.html#category-autoclear) ?>
+                          cookie.name = new RegExp(cookie.name);
+                      }
+                  });
+              }
+          }
+      }
 
-    <?php if ( apply_filters( 'air_cookie\styles\set_max_width', true ) ) : ?>
-      var cookieconsent_element = document.querySelector('div#cc_div div#cm');
-      if( typeof( cookieconsent_element ) != 'undefined' && cookieconsent_element != null ) {
-        cookieconsent_element.style = 'max-width: 30em;';
+      <?php // Fixes: issues with embedded content when changing consent. ?>
+      function checkIframeConsent() {
+        if ( typeof manager !== 'undefined' ) {
+            if ( ! CookieConsent.getCookie( 'categories' ).includes('embeds') ) {
+              manager.rejectService('all');
+            }
+            else {
+              manager.acceptService('all');
+            }
+          }
       }
+
     <?php endif; ?>
 
+      <?php // Add functions to handle changes ?>
+      const ccOnChanges = {
+        onFirstConsent: () => {
+          ccOnAccept();
+        },
+
+        onConsent: () => {
+          checkIframeConsent();
+        },
+
+        onModalShow: () => {
+          <?php if ( apply_filters( 'air_cookie\styles\set_max_width', true ) ) : ?>
+            var cookieconsent_element = document.querySelector('div#cc-main div.cm');
+            if( typeof( cookieconsent_element ) != 'undefined' && cookieconsent_element != null ) {
+              cookieconsent_element.style = 'max-width: 40em;';
+            }
+          <?php endif; ?>
+        },
+
+        onChange: () => {
+          ccOnChange();
+        }
```

Comment:
onFirstConsent, onConsent and onChange probably don't need to be inside a arrow function and can be passed as arguments.

`onChange: ccOnChange`.